components/tools/Filters.js

```diff
@@ -47,6 +47,38 @@ export default function Filters({setOpenFilter}) {
     setOpenFilter(false)
   }
   
+  // function to clear selected filters
+  const handleClickOption = (e, option, checkedDomain, setDomain) => {
+    let tempValueArray = [...checkedDomain]
+    let index = checkedDomain.indexOf(option)
+    if(index>-1){
+      tempValueArray.splice(index, 1);
+    }
+    setDomain(tempValueArray)
+  }
+
+  // function to display filtered domain
+  const FilterDisplay = (props) => {
+    return (
+      <div className="max-w-lg flex gap-2 flex-wrap p-2 duration-200 delay-150">
+        {props.checkedDomain.map((items, index) => {
+          return (
+            <div
+              key={index}
+              className={twMerge(
+                `hover:bg-neutral-600 border border-gray-600 text-gray-600 p-1 pb-0 rounded-2xl flex gap-1 cursor-pointer items-start `
+              )}
+            >
+              <div className="text-xs -mt-[1px] mb-[1px]">{items}</div>
+              <button className="hover:bg-gray-100 p-0.5 rounded-full" onClick={(e) => handleClickOption(e, items, props.checkedDomain, props.setDomain)}>
+              <img src="/img/illustrations/icons/close-icon.svg" width="10" /></button>
+            </div>
+          );
+        })}
+      </div>
+    );
+  };
```

Comment:
Kindly make a separate file for this component. You should dispatch the state change functions if required in order to get values changed using the close button. There shouldn't be any component inside a component.